File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    heapq.heapify(scoville) 
    answer = 0
    
    while True:
        if len(scoville) <= 1 and scoville[0] < K:
            answer = -1
            break
        if scoville[0] >= K:
            break
        
        first = heapq.heappop(scoville)
        second = heapq.heappop(scoville)
        
        heapq.heappush(scoville, first + second * 2)
        answer += 1
        
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        
        for (int s: scoville) {
            heap.add(s);
        }
        
        int answer = 0;
        while (true) {
            if (heap.size() <= 1 && heap.peek() < K) {
                answer = -1;
                break;
            }
            
            if (heap.peek() >= K) {
                break;
            }
            
            int first = heap.poll();
            int second = heap.poll();
            heap.add(first + second * 2);
            answer += 1;
        }
        
        return answer;
    }
}